app/view/twig/_nav/_macros.twig

```diff
@@ -8,8 +8,7 @@
  # @param bool    wide
  # @param array   subitems
  #}
-{% macro submenu(icon, label, popoveritems, active, wide, subitems, force_submenu) %}
-
+{% macro submenu(icon, label, popoveritems, active, wide, subitems, force_submenu, singleton) %}
```

Comment:
Don't move _more_ logic into Twig, this whole macro needs desperately to :boom: :koala: :fire: 

_**Hint:**_ Have a look at what `subitems` contains